src/login/oidc/oidcHandlers/LegacyImplicitFlowOidcHandler.ts

```diff
@@ -38,15 +40,22 @@ export default class LegacyImplicitFlowOidcHandler implements IOidcHandler {
     @inject("sessionInfoManager")
     private sessionInfoManager: ISessionInfoManager,
     @inject("redirector") private redirector: IRedirector,
-    @inject("dpopHeaderCreator") private dpopHeaderCreator: IDpopHeaderCreator
+    @inject("dpopClientKeyManager")
+    private dpopClientKeyManager: IDpopClientKeyManager,
+    @inject("dpopHeaderCreator") private dpopHeaderCreator: IDpopHeaderCreator,
+    @inject("storageUtility") private storageUtility: IStorageUtility
   ) {}
 
   async canHandle(oidcLoginOptions: IOidcOptions): Promise<boolean> {
     return !!(
       oidcLoginOptions.issuerConfiguration.grantTypesSupported &&
       oidcLoginOptions.issuerConfiguration.grantTypesSupported.indexOf(
         "implicit"
-      ) > -1
+      ) > -1 &&
+      // Escape hatch to detect that we are talking to NSS and not the id broker
+      oidcLoginOptions.issuerConfiguration.grantTypesSupported.indexOf(
+        "urn:ietf:params:oauth:grant-type:jwt-bearer"
```

Comment:
This is a very brittle comparison.

Reply:
Ideally I'd like to figure out what goes wrong with the auth code flow on NSS, and then this wouldn't be necessary. If the need to use implicit flow on NSS and auth code flow on the broker lasts, a better design (read: any design really) will be needed to replace this.

Reply:
Acknowledged in e65ae11bb5e81297c1fbc3e19e0219f4d3b5282d